pkg/review/commit_review_status.go

```diff
@@ -192,12 +233,42 @@ func (fn *CommitApprovalDoFn) ProcessElement(ctx context.Context, commit Commit,
 	emit(commitReviewStatus)
 }
 
+// ProcessElement is a DoFn implementation that takes a CommitReviewStatus
+// and populates its breakGlassIssue field (if necessary) and then emits it
+// using the given emit function. The process only searches for break glass
+// issues for commits that do not have the status GithubPRApproved.
+func (fn *BreakGlassIssueDoFn) ProcessElement(ctx context.Context, commitReviewStatus CommitReviewStatus, emit func(status CommitReviewStatus)) {
+	if emit == nil {
+		panic("A nil emit function was passed in. Please check beam pipeline setup.")
+	}
+	logger := logging.FromContext(ctx)
+	logger.InfoContext(ctx, "processing commitReviewStatus", "commitReviewStatus", commitReviewStatus)
+	if commitReviewStatus.ApprovalStatus != GithubPRApproved {
+		// if the commit does not have proper approval, we check if there was a
+		// break glass issue opened by the author during the timeframe they
+		// submitted the commit.
+		breakGlassIssues, err := fn.BreakGlassIssueFetcher.getBreakGlassIssues(ctx, commitReviewStatus.Author, commitReviewStatus.Timestamp)
+		if err != nil {
+			// We should only get transient style errors from BigQuery
+			// (e.g. network is down etc.). So, we can just log the error and then
+			// drop this CommitReviewStatus from the pipeline. It will then get
+			// retried on the next run of the pipeline.
+			logger.ErrorContext(ctx, "failure when trying to get break glass issue: %v", err)
+			return
+		}
+		commitReviewStatus.BreakGlassURLs = mapSlice(breakGlassIssues, func(issue breakGlassIssue) string {
+			return issue.HTMLURL
+		})
+	}
+	emit(commitReviewStatus)
```

Comment:
Can/should this be backgrounded (`go emit(commitReviewStatus)`)? Or do we want to block during the emission?

Reply:
From all the examples int the Apache Beam documentation the `emit` function is directly called and not spun off into a separate goroutine. I couldn't find anything in the docs saying this was a requirement, but I think keeping it as is (a potential blocking call) is fine.